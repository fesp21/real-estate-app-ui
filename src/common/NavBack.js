import React, {Component, PropTypes} from "react";
import {View, StyleSheet, StatusBar, Text,TouchableHighlight} from "react-native";
import Colors from './../common/Colors';

import {
  withNavigation,
} from '@exponent/ex-navigation';

@withNavigation
export default class NavBack extends Component {

  static propTypes = {
    // style:PropTypes.node,
    text:PropTypes.string,
    navigator:PropTypes.object.isRequired,
    icon:PropTypes.string
  };

  goBack() {
    this.props.navigator.pop();
  }

  render() {
    const {style,text,icon} = this.props;
    return (
      <View style={styles.container}>
        <TouchableHighlight onPress={()=>{this.goBack()}} underlayColor="transparent">
            {icon ? <Ionicons name={icon} size={30} color="#2c2d30" style={[styles.icon]}/> : <Text style={[styles.title,style]}>{text}</Text> }
        </TouchableHighlight>
      </View>
    );
  }
}

const styles =  StyleSheet.create({
  container : {
    flex:1,
    justifyContent:'center',
    flexDirection:'row',
    alignItems:'center',
    paddingLeft:15,
    paddingRight:10
  },
  title:{
    color:Colors.tomato,
    fontSize:15,
  },
  icon: {
    width:30,
    height:30,
    alignSelf:'center',
    color:Colors.tomato,
  }
});