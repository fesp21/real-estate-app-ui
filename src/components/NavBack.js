import React from "react";
import {
  View,
  StyleSheet,
  StatusBar,
  Text,
  TouchableHighlight
} from "react-native";
import colors from "./../common/colors";
import Ionicons from "react-native-vector-icons/Ionicons";

import { withNavigation } from "@exponent/ex-navigation";

@withNavigation
export default class NavBack extends React.Component {
  static propTypes = {
    // style:View.propTypes.style,
    text: React.PropTypes.string,
    navigator: React.PropTypes.object.isRequired,
    icon: React.PropTypes.string
  };

  goBack() {
    this.props.navigator.pop();
  }

  render() {
    const { style, text, icon } = this.props;
    return (
      <View style={styles.container}>
        <TouchableHighlight
          onPress={() => {
            this.goBack();
          }}
          underlayColor="transparent"
        >
          {icon
            ? <Ionicons
                name={icon}
                size={30}
                color="#2c2d30"
                style={[styles.icon, style]}
              />
            : <Text style={[styles.title, style]}>{text}</Text>}
        </TouchableHighlight>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    flexDirection: "row",
    alignItems: "center",
    paddingLeft: 15,
    paddingRight: 10
  },
  title: {
    color: colors.tomato,
    fontSize: 15
  },
  icon: {
    width: 30,
    height: 30,
    alignSelf: "center",
    color: colors.tomato
  }
});
