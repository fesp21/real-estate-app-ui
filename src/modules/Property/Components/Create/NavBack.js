import React from 'react';
import { View, StyleSheet, Text, TouchableHighlight } from 'react-native';
import Colors from './../../../../common/Colors';
import Ionicons from 'react-native-vector-icons/Ionicons';
import isNull from 'lodash/isNull';

export default class NavBack extends React.Component {

  goBack() {
    this.props.emitter.emit('goBack');
  }

  render() {
    const { style, text, icon, stage } = this.props;
    return (
      <View style={{ flex: 1 }}>
        {!isNull(stage) && stage > 1 &&
        <TouchableHighlight
          style={styles.container}
          onPress={() => { this.goBack(); }} underlayColor="transparent"
        >
          {icon ? <Ionicons name={icon} size={33} color="white" style={[styles.icon]} /> :
          <Text style={[styles.title, style]}>{text}</Text> }
        </TouchableHighlight>
        }
      </View>
    );
  }
}

NavBack.propTypes = {
  text: React.PropTypes.string,
  icon: React.PropTypes.string,
  // style: React.PropTypes.object,
  stage: React.PropTypes.number,
  emitter: React.PropTypes.object.isRequired,
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: 8,
    paddingRight: 10,
  },
  title: {
    color: Colors.tomato,
    fontSize: 15,
  },
  icon: {
    width: 13,
    height: 33,
    alignSelf: 'center',
    color: Colors.white,
  },
});

