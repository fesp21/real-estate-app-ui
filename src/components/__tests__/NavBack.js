import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import NavBack from "./../NavBack";

const navigator = {
  pop:jest.fn()
};

test('renders null', () => {
  const tree = renderer.create(
    <NavBack
      text="back"
      navigator={navigator}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});