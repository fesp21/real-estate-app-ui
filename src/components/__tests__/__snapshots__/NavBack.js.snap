exports[`test renders null 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingLeft": 15,
      "paddingRight": 10,
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#d60c3f",
            "fontSize": 15,
          },
          undefined,
        ]
      }>
      back
    </Text>
  </View>
</View>
`;
